[BUG] Error C4996: 'Py_UNICODE': deprecated in 3.13 - on cp313-win32.

### Describe the bug

I have a Cython project that errors with error C4996: 'Py_UNICODE': deprecated in 3.13 only when building on Python 3.13 on win32. The same code builds with Python 3.13 on Windows with amd64 and on ubunto-latest and macos-latest. 

The project does not explicitly use Py_UNICODE but does use array.array('B') and I see [from ](url)https://github.com/cython/cython/issues/5982#issuecomment-1955195289 there may be connection. 

Error:
https://github.com/microsoft/VisualTrueType/actions/runs/12700069355/job/35402026392

Thanks



### Code to reproduce the behaviour:

```cython
# example code
```


### Expected behaviour

_No response_

### OS

Windows win32. 

### Python version

Python 3.13

### Cython version

current

### Additional context

_No response_

--- a/Cython/Utility/arrayarray.h
+++ b/Cython/Utility/arrayarray.h
@@ -51,7 +51,10 @@ struct arrayobject {
         unsigned short *as_ushorts;
         // Don't use Py_UNICODE ourselves in the union. This avoids deprecation warnings 
         // for anyone who uses array.array but doesn't use this field.
-        Py_DEPRECATED(3.13) wchar_t *as_pyunicodes;
+        #if PY_VERSION_HEX >= 0x030d0000
+        Py_DEPRECATED(3.13) 
+        #endif
+            wchar_t *as_pyunicodes;
         void *as_voidptr;
     } data;
     Py_ssize_t allocated;

--- a/Cython/Utility/arrayarray.h
+++ b/Cython/Utility/arrayarray.h
@@ -49,7 +49,9 @@ struct arrayobject {
         long long *as_longlongs;
         short *as_shorts;
         unsigned short *as_ushorts;
-        Py_UNICODE *as_pyunicodes;
+        // Don't use Py_UNICODE ourselves in the union. This avoids deprecation warnings 
+        // for anyone who uses array.array but doesn't use this field.
+        Py_DEPRECATED(3.13) wchar_t *as_pyunicodes;
         void *as_voidptr;
     } data;
     Py_ssize_t allocated;

